### ISSUE TITLE
Investigate interoperability with Sukritact's Simple UI Adjustments

### ISSUE THREAD

**Issue by the-m4a** [civfanatics/CQUI_Community-Edition#33]
**Is your feature request related to a problem? Please describe.**
A number of folks that have subscribed to this mod also have subscribed to Sukritact's Simple UI adjustments.  The two mods do not coexist at the moment and considering Sukritact typically produces high-quality mods, I would like to see if we can make the two co-exist.

We should not integrate that mod unless we explicitly receive permission to do so.

https://steamcommunity.com/sharedfiles/filedetails/?id=939149009

**Describe the solution you'd like**
We are able to use both mods simultaneously.

**Describe alternatives you've considered**
We can mark our mod as incompatible by adding the id of the Simple UI adjustments mod to our Block list.

**Additional context**
The mods nearly work together as is, as it appears there's conflicts with the city banner (simple UI adjustments shows a tooltip on mouseover) and closing the city screen requires using the ESC key as the [X] does not seem to respond to the click.

**Comment by the-m4a** [comment]
I started looking at this, it's not going to be easily done because Sukritact does file replacing rather than the extending - at least for CityPanel and maybe CityBanner..

I would LOVE to have the stuff that happens when you mouse-over the city banner, though.  Would love to figure a way to get that functionality incorporated.

**Comment by Infixo** [comment]
> I would LOVE to have the stuff that happens when you mouse-over the city banner, though. Would love to figure a way to get that functionality incorporated.

What exactly happens? I don't use this mod.

**Comment by the-m4a** [comment]
it turns out this mod does do this, or at least something similar to it, I just forgot that was the case when I made this comment

![image](https://user-images.githubusercontent.com/8787640/85238677-6b293700-b3e4-11ea-97e9-02ce72591266.png)

**Comment by Infixo** [comment]
How to make some of SukUI features work with CQUI (specifically Religion tooltip and Production tooltip).
1. Make a local copy of Suk's mod.
2. Replace .modinfo and 1 .lua file with the ones attached here.
[SukUlwithCQUI.zip](https://github.com/civfanatics/CQUI_Community-Edition/files/4825168/SukUlwithCQUI.zip)

If you want to also have nicely formatted tooltips for city yields, then you need to insert this piece of code into citypanel.lua from CQUI at line 648 (after the section commented as   -- Set the Yield checkboxes based on the game state).

```
-- Infixo SukUI integration
if Modding.IsModActive("19F1AFFF-17EF-5E77-6287-53D559A5981B") then
  local toPlusMinus = function(value) return Locale.ToNumber(value, "+#,###.#;-#,###.#") end
  LuaEvents.SetSuk_YieldTooltip(Controls.CultureGrid,    data.YieldFilters[YieldTypes.CULTURE],    toPlusMinus(data.CulturePerTurn),    data.CulturePerTurnToolTip);
  LuaEvents.SetSuk_YieldTooltip(Controls.FaithGrid,      data.YieldFilters[YieldTypes.FAITH],      toPlusMinus(data.FaithPerTurn),      data.FaithPerTurnToolTip);
  LuaEvents.SetSuk_YieldTooltip(Controls.FoodGrid,       data.YieldFilters[YieldTypes.FOOD],       toPlusMinus(totalFood),              realFoodPerTurnToolTip); -- different here
  LuaEvents.SetSuk_YieldTooltip(Controls.GoldGrid,       data.YieldFilters[YieldTypes.GOLD],       toPlusMinus(data.GoldPerTurn),       data.GoldPerTurnToolTip);
  LuaEvents.SetSuk_YieldTooltip(Controls.ProductionGrid, data.YieldFilters[YieldTypes.PRODUCTION], toPlusMinus(data.ProductionPerTurn), data.ProductionPerTurnToolTip);
  LuaEvents.SetSuk_YieldTooltip(Controls.ScienceGrid,    data.YieldFilters[YieldTypes.SCIENCE],    toPlusMinus(data.SciencePerTurn),    data.SciencePerTurnToolTip);
end

```

The tooltip presents exactly the same information as original one, it is just nicely formatted.

Other features are disabled for various reasons e.g. showing worked plots with mouse over conflicts directly with CQUI feature, can't use both at the same time. That is why the change in .lua is needed in Suk's mod - to disable this feature.

**Comment by Sukritact** [comment]
Which files that I'm replacing are conflicting in particular? It should be possible to set it up so that my mod will NOT replace those files while this mod is active via criteria.

**Comment by Infixo** [comment]
Yup, I think it possible to make them working together out-of-the-box.
CityPanel.lua and .xml, CQUI has its own versions, so probably leaving low LoadOrder would be enough, then CQUI will overwrite them.
CityBannerManager_Suk_UI - this needs to have LoadOrder > 200 and mouse-over disabled when CQUi is active, easy to do.

StatusMessagePanel_SukUI threw some error message, but I didn't dig deeper.

**Comment by Sukritact** [comment]
Just updated Simple UI Adjustments. It looks like it should be compatible with CQUI out of the box now, though I haven't tested it extensively yet.

**Comment by the-m4a** [comment]
Y'all are awesome, thank you!  Much appreciated, I will look into it as well.  Need to look into whether or not the current patch broke things later on this evening :/

**Comment by Infixo** [comment]
@Sukritact 
CityBanner was refactored and there is no longer CityBanner.lua in the mod. Atm religion and production tooltips won't work.

CityBannerManager_Suk_UI.lua
- it should have LoadOrder >200 in ReplaceUIScript to overwrite cqui

it should include the proper base file like this:

g_IS_CQUI = Modding.IsModActive("1d44b5e7-753e-405b-af24-5ee634ec8a01");
<add xp2 check>
<add xp1 check

if not g_IS_CQUI then
  include("CityBannerManager.lua");
else
  if xp1 or xp2 then 
    include("citybannermanager_CQUI_expansions");
  else
    include("citybannermanager_CQUI_basegame");
end

**Comment by Sukritact** [comment]
Thanks, I'll see if I can integrate the changes in a few days

**Comment by Sukritact** [comment]
Apologies for the delay. Simple UI Adjustments has been updated to fix the above issue.

**Comment by the-m4a** [comment]
Thank you again!
